pywren/wrenhandler.py

```diff
@@ -211,67 +204,13 @@ def generic_handler(event, context_dict):
             data_key_size = get_key_size(s3_client, s3_bucket, data_key)
         if not event['use_cached_runtime']:
             subprocess.check_output("rm -Rf {}/*".format(RUNTIME_LOC), shell=True)
-        func_key_size = get_key_size(s3_client, s3_bucket, func_key)
+
 
         free_disk_bytes = free_disk_space("/tmp")
-        if (func_key_size + data_key_size) > (free_disk_bytes - TMP_MIN_FREE_SPACE_BYTES):
-            raise Exception("ARGS_TOO_BIG",
-                            "data + func too large {:3.1f} MB (data={:3.1f}MB, func={:3.1f}MB)," \
-                            " free space on worker is {:3.1f} MB, " \
-                            " need at least {:3.1f} MB of free space headroom to run"\
-                            .format((func_key_size + data_key_size)/1e6,
-                                    data_key_size/1e6, func_key_size/1e6,
-                                    free_disk_bytes/1e6,
-                                    TMP_MIN_FREE_SPACE_BYTES/1e6))
-        # get the input and save to disk
-        # FIXME here is we where we would attach the "canceled" metadata
-        s3_transfer.download_file(s3_bucket, func_key, func_filename)
-        func_download_time = time.time() - start_time
-        response_status['func_download_time'] = func_download_time
-
-        logger.info("func download complete, took {:3.2f} sec".format(func_download_time))
-
-        if data_byte_range is None:
-            s3_transfer.download_file(s3_bucket, data_key, data_filename)
-        else:
-            range_str = 'bytes={}-{}'.format(*data_byte_range)
-            dres = s3_client.get_object(Bucket=s3_bucket, Key=data_key,
-                                        Range=range_str)
-            data_fid = open(data_filename, 'wb')
-            data_fid.write(dres['Body'].read())
-            data_fid.close()
-
-        data_download_time = time.time() - start_time
-        logger.info("data data download complete, took {:3.2f} sec".format(data_download_time))
-        response_status['data_download_time'] = data_download_time
-
-        # now split
-        d = json.load(open(func_filename, 'r'))
+        response_status['free_disk_bytes'] = free_disk_bytes
+        # clean up for modules
         shutil.rmtree(PYTHON_MODULE_PATH, True) # delete old modules
```

Comment:
should we also move this inside the jobrunner ? I'm not sure it matters but it'll make the behavior same as before (i.e. download func and if that succeeds then do rmtree)

Reply:
yeah good catch will do